gen-ai/orchestrator-server/src/main/python/server/src/gen_ai_orchestrator/services/langchain/rag_chain.py

```diff
@@ -312,3 +337,46 @@ def get_rag_debug_data(
         answer=response['answer'],
         duration=rag_duration,
     )
+
+
+def check_guardrail_output(guardrail_output: dict) -> bool:
+    """Checks if the guardrail detected toxicities.
+
+    Parameters
+    ----------
+    guardrail_output: dict
+
+    Returns
+    -------
+    Returns True if nothing is detected, raises an exception otherwise.
+    """
+    if guardrail_output['output_toxicity']:
+        raise HTTPException(
+            status_code=451,
+            detail=f"Toxicity detected in LLM output ({','.join(guardrail_output['output_toxicity_reason'])})",
+        )
+    return True
+
+
+def add_compressor(
+    retriever: VectorStoreRetriever, compressor_settings: BaseCompressorSetting
+) -> ContextualCompressionRetriever:
+    """Adds a compressor to the retriever.
+
+    Parameters
+    ----------
+    retriever : VectorStoreRetriever
+        Base retriever.
+    compressor_settings : BaseCompressorSetting
+        Compressor settings.
+
+    Returns
+    -------
+    New retriever with compressing feature.
+    """
```

Comment:
garde la même manière de décrire les fonctions, pour homogénéisé le doc

```suggestion
    """
    Adds a compressor to the retriever.
    Args:
        retriever : the Base retriever
        compressor_settings : the compressor settings
    Returns:
        New retriever with compressing feature.
    """
```